src/components/editor/menu/DiagramOptions.tsx

```diff
@@ -44,7 +47,8 @@ const DiagramOptions = ({
         {onCutSetAnalysis && (
           <Tooltip title={`${t("diagramSidePanel.cutsetToggleToolTip")}`}>
             <IconButton color="primary" onClick={onCutSetAnalysis} size="large">
-              <RouteIcon />
+              {rendering && <CircularProgress size={24} />}
```

Comment:
IMO, it is clearer/simpler with `{rendering ? <CircularProgress size={24} /> : <RouteIcon />}`. Also, it is not the case here, but it can have some drawbacks when using numbers (see react [doc](https://www.crocoder.dev/blog/react-conditional-rendering/)) (e.g. `0 && <CircularProgress />`). So, in general it can be conisdered good practice to user ternary operators rather than operators when possible.

Reply:
Agree, fixed